### ISSUE TITLE
DualShock 4 Slim Not Recognized by Steam Or Any Steam Games, This Can Be Fixed Very Quickly

### ISSUE THREAD

**Issue by Alexay** [ValveSoftware/steam-for-linux#4683]
#### Your system information

* Steam client version: 1:1.0.0.48-1ubuntu3
* Distribution (e.g. Ubuntu): Ubuntu 16.04.1 x64
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
DualShock 4 Slim is a slight revision of the normal DualShock 4 controller. The device name is slightly different and so is the device ID.

When I connect the controller by USB or pair it by Bluetooth Steam doesn't recognize it at all (when you press the PlayStation button nothing happens as opposed to the Steam window opening like with the standard DualShock 4)

I know for a fact that the controller works (at least through USB) in Ubuntu because http://html5gamepad.com/ correctly recognizes the gamepad and all the button manipulations.

My `lsusb` sees the controller as:
`Bus 003 Device 008: ID 054c:09cc Sony Corp. `

When connecting the controller through USB `dmesg` outputs the following:
```
[583905.131514] usb 3-10: new full-speed USB device number 9 using xhci_hcd
[583905.260575] usb 3-10: New USB device found, idVendor=054c, idProduct=09cc
[583905.260577] usb 3-10: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[583905.260578] usb 3-10: Product: Wireless Controller
[583905.260579] usb 3-10: Manufacturer: Sony Interactive Entertainment
[583905.260649] usb 3-10: ep 0x84 - rounding interval to 32 microframes, ep desc says 40 microframes
[583905.260651] usb 3-10: ep 0x3 - rounding interval to 32 microframes, ep desc says 40 microframes
[583905.271098] input: Sony Interactive Entertainment Wireless Controller as /devices/pci0000:00/0000:00:14.0/usb3/3-10/3-10:1.3/0003:054C:09CC.0049/input/input125
[583905.271196] hid-generic 0003:054C:09CC.0049: input,hidraw1: USB HID v1.11 Gamepad [Sony Interactive Entertainment Wireless Controller] on usb-0000:00:14.0-10/input3
```

Notice that the device name is `"Sony Interactive Entertainment Wireless Controller"` as opposed to `"Sony Computer Entertainment Wireless Controller"`, which is the device name for the standard DualShock 4. This is probably the root cause of the problem. A quick addition of the device name/id to the Steam client device scanner should do the trick.

#### Steps for reproducing this issue:

1. Connect the DualShock 4 Slim controller through USB;
2. Press the PlayStation button;
3. Nothing happens.

**Comment by ProfessorKaos64** [comment]
>Notice that the device name is "Sony Interactive Entertainment Wireless Controller" as opposed to "Sony Computer Entertainment Wireless Controller", which is the device name for the standard DualShock 4. This is probably the root cause of the problem. A quick addition of the device name/id to the Steam client device scanner should do the trick.

I would think this is not so much a Steam problem, but an xpad issue. It likely isn't added to xpad.c itself. You cannot rely on an HTML5 website for the joystick verification. Rather, qjoypad/udev/antimicro/etc. should verify details. Does this work in any game outside of Steam when the prior gamepad did?

**Comment by Tele42** [comment]
This should be [fixed upstream](https://hg.libsdl.org/SDL/rev/1b7719425a6d) and will be included when libsdl 2.0.5 is widely deployed.

**Comment by Plagman** [comment]
As @Tele42 mentioned this'll be fixed naturally soon, but Steam's upcoming native PS4 support will also handle PS4 Slim controllers and let you program them in the same way you could a Steam Controller.
